frontend/src/app/home/Pet/Collection.tsx

```diff
@@ -0,0 +1,109 @@
+"use client";
+
+import { Pet } from ".";
+import { useCallback, useState, useEffect } from "react";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { Network, Provider } from "aptos";
+
+export const provider = new Provider(Network.TESTNET);
+
+export interface CollectionProps {
+  pet: Pet;
+}
+
+export type Collection = {
+  collection_id: string;
+  collection_name: string;
+  creator_address: string;
+  uri: string;
+  current_supply: any;
+};
+
+export function Collection({ pet }: CollectionProps) {
+  const { account, network } = useWallet();
+  const [collection, setCollection] = useState<Collection>();
+  const [collectionHolders, setCollectionHolders] = useState<string[]>();
+
+  const fetchCollection = useCallback(async () => {
+    if (!account?.address) return;
+
+    const tokenDataResponse = await provider.getTokenData(pet.address, {
+      tokenStandard: "v2",
+    });
+
+    const collectionResponse =
+      tokenDataResponse?.current_token_datas_v2[0].current_collection!;
+
+    const getAllCollectionHoldersGql = {
+      query: `
+        query MyQuery($collection_id: String) {
+          current_collection_ownership_v2_view(
+            where: { collection_id: { _eq: $collection_id } }
+          ) {
+            owner_address
+          }
+        }
+      `,
+      variables: {
+        collection_id: collectionResponse.collection_id,
```

Comment:
can collection_id be queried from a view function in the contract? it's never gonna change so doesn't seem clean to query it every time with the token address.